### PR TITLE
local-build: always use DEFSERVICEOFFLOAD for CoCo

### PR DESCRIPTION
**Pull request by fitzthum**
DEFSERVICEOFFLOAD controls whether images are pulled inside the guest. This should always be set for CoCo, not just when we use MEASURED_ROOTFS.

Fixes: #7350

### CODE REVIEW

**Review by UnmeshDeodhar — APPROVED** [review]
LGTM, thanks @fitzthum!

**Review by wainersm — APPROVED** [review]
Good catch @fitzthum ! When I reviewed the original pull request I found a little strange the export inside the MEASURED_ROOTFS block but I never thought it would disable on the other cases!

**Comment by fidencio** [comment]
Added the `do-not-merge` label just to make sure all the TEEs tests pass before it gets merged.

**Review by stevenhorsman — APPROVED** [review]
LGTM assuming that all the tests pass

**Comment by fidencio** [comment]
CLH TDX passed but didn't return back: http://10.112.240.228:8080/job/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/972/console

I'll leave the `do-not-merge`label till the SEV CI passed.  However looks at this later on, please, feel free to remove when it happens.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @fitzthum!

**Review by arronwy — APPROVED** [review]
Thanks @fitzthum , TDX CI passed, but did not report back:
http://10.112.240.228:8080/view/CCv0%20PR%20Triggered/job/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/972/
http://10.112.240.228:8080/view/CCv0%20PR%20Triggered/job/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/1033/

**Comment by stevenhorsman** [comment]
We're getting closer on the SEV tests. Initially there was a test failure, so I re-queued and now all the tests pass, but the cleanup fails:
```
09:47:47 bash: /home/jenkins/workspace/kata-containers-CCv0-ubuntu-20.04_sev-x86_64-CC_SEV_CRI_CONTAINERD_K8S-PR/go/src/github.com/kata-containers/tests/integration/kubernetes/cleanup_bare_metal_env.sh false: No such file or directory
09:47:47 [cleanup_env.sh:55] INFO: Check no kata processes are left behind after reseting kubernetes
09:47:48 [cleanup_env.sh:58] INFO: Checks that pods were not left
09:47:48 16dbd748ad28058fb1b7d9ccd9a0bb3f1e2f9543fdf78cfc6ad21456dcaed677
09:47:48 26d79274a1105b0167d72d1f2da9c97d0ae443e5f127545d3c7d8ea3e1105be5
09:47:48 47f8f5b0a1a964bb1b3514d80fe9b29d54091c975e8737c8e74bc58abac82db6
09:47:48 ERROR: 3 pods left and found at /run/vc/sbs
```
I'll have another go, but it might need someone with access to tidy things up?

**Comment by stevenhorsman** [comment]
Third time lucky for SEV. Given that both TDX passed too, I think this is ready to be merged.

**Pull request merged**
